app/src/main/java/com/example/superheroapi/MainActivity.kt

```diff
@@ -56,47 +56,62 @@ class MainActivity : ComponentActivity() {
 }
 
 @Composable
-fun SuperHero(hero: SuperHero,
-              searchMethod: (name: String) -> SuperHeroSearch,
-              getMethod: (id: String) -> SuperHero,
-              modifier: Modifier = Modifier) {
-    Scaffold(
-        topBar = { SearchField(modifier.padding(bottom = 8.dp)) },
-        bottomBar = {},
-        modifier = modifier.padding(start = 8.dp, end = 8.dp)
-    ) {
-        SuperHeroCard(
-            hero = hero,
-            modifier = modifier
-                .padding(it)
-                .padding(bottom = 8.dp)
+fun SuperHeroScreen(
+    modifier: Modifier = Modifier,
+    uiState: SuperHeroUiState,
+    getMethod: (id: String) -> Unit,
+) {
+    Column {
+        SearchField(
+            modifier = modifier.padding(bottom = 8.dp),
+            getMethod = { input -> getMethod(input) }, // the "input" parameter can be given any name you want
         )
+        when (uiState) {
+            is SuperHeroUiState.Success -> {
+
+                SuperHeroCard(
+                    hero = uiState.hero,
+                    modifier = modifier
+                        .padding(bottom = 8.dp)
+                )
+            }
+
+            SuperHeroUiState.Error ->
+                Text(text = "Error")
+
+            SuperHeroUiState.Loading ->
+                Text(text = "Loading")
+        }
     }
```

Comment:
Reworked where the ui state is handled in the Composable to make the screen work properly (the Loading text is now displayed below the search bar instead of previously, where the search bar was not displayed if the ui state was set to Loading or Error).